mapslang/Makefile

```diff
@@ -1,27 +1,56 @@
-# Project: adikted
-
-#To create a debug executable, run with the command:
-# make DEBUG
-
-#ON_LINUX = 1
-
-ifneq (,$(ON_LINUX))
+# Project: mapslang
+# Makefile created by KeeperFX/dkfans team
+
+# Instructions:
+# To create the Windows executable run:
+# make all
+# To create the linux executable run:
+# make all BUILD_FOR_LINUX=1
+# To create a debug executable, run with the command:
+# make DEBUG (THIS IS NOT IMPLEMENTED YET)
+# To clean up the built files (objects, library, and resource file) run:
+# make clean
+
+# Tested and working with:
+# - Windows : MSYS2 (MinGW 32-bit) - make
+# - Windows : MSYS2 (MinGW 32-bit) - mingw32-make
+# - Windows : MinGW (MSYS) - make
+
+# Executable files extension on host environment
+ifneq (,$(findstring Windows,$(OS)))
+  CROSS_EXEEXT = .exe
+else
+  CROSS_EXEEXT =
   CROSS_COMPILE = i686-w64-mingw32-
-  SLANG_HOST=mingw32
+  SLANG_HOST = mingw32
 endif
-
-SLANG_URL = https://www.jedsoft.org/snapshots/slang-pre2.3.3-24.tar.gz
-SLANG_NAME=slang-pre2.3.3-24
-
-CC                = $(CROSS_COMPILE)gcc
-CPP                = $(CROSS_COMPILE)g++
-CFLAGS                          = -c -O -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1
-CFLAGS-DEBUG                    = -c -g -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1
-LINK                = $(CROSS_COMPILE)gcc
-WINDRES                         = $(CROSS_COMPILE)windres
-LFLAGS                          = 
-LFLAGS-DEBUG                    = 
-LIBS                = -L. -L../libadikted -L../lib/$(SLANG_NAME)/src/gw32objs -lslang -ladikted
+# Executable files extension on target environment (
+#BUILD_FOR_LINUX = 1
+ifneq (,$(BUILD_FOR_LINUX))
+  EXEEXT =
+  OTHEREXEEXT = .exe
+  LIBEXT = .so
+  OTHERLIBEXT = .dll
+else
+  EXEEXT = .exe
+  OTHEREXEEXT =
+  LIBEXT = .dll
+  OTHERLIBEXT = .so
+endif
+# Names of utility commands
+SLANG_URL          = https://www.jedsoft.org/snapshots/slang-pre2.3.3-24.tar.gz
+SLANG_NAME         = slang-pre2.3.3-24
+CPP                = $(CROSS_COMPILE)g++$(CROSS_EXEEXT) -std=c18
+CC                 = $(CROSS_COMPILE)gcc$(CROSS_EXEEXT) -std=c18 -Wall -Wextra -Wno-pedantic -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare
+CFLAGS             = -c -O -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1 -march=i486
```

Comment:
`-march=i486`

Another ignorant check by me, is this needed? And is it correct that it is different to the MMX ones in `libadikted/Makefile`?

Reply:
As with my prior comment, removing it would make it use a default value of `i686`. We would have to reach out to mefistotelis and ask him why that was kept at `i486`, perhaps the slang dependency?

Reply:
Both `-march=i486` and `-march=pentium-mmx -mmmx` are now set to the variable `PLATFORMFLAGS` in either Makefile. So if these ever need to be modified, it is even easier now.